File: server/Services/FavoritesService.cs

namespace allspice_dotnet_fullstack.Services;

public class FavoritesService
{
  private readonly FavoritesRepository _favoritesRepository;

  public FavoritesService(FavoritesRepository favoritesRepository)
  {
    _favoritesRepository = favoritesRepository;
  }

  internal FavoriteRecipe CreateFavorite(Favorite favoriteData)
  {
    FavoriteRecipe favorite = _favoritesRepository.CreateFavorite(favoriteData);
    return favorite;
  }
}

File: client/src/services/FavoritesService.js
import { logger } from "@/utils/Logger.js"
import { api } from "./AxiosService.js"
import { Favorite } from "@/models/Favorite.js"
import { AppState } from "@/AppState.js"

class FavoritesService {
  async getMyFavorites() {
    const response = await api.get('/account/favorites')
    logger.log('Got Favorites!', response.data)
    const favorites = response.data.map(pojo => new Favorite(pojo))
    AppState.favorites = favorites
  }

  async createFavorite(favoriteData) {
    logger.log('favorite data', favoriteData)
    const response = await api.post('api/favorites', favoriteData)
    logger.log('created favorite!', response.data)
    const favorite = new Favorite(response.data)
    logger.log('favorite', favorite)
    return favorite
  }

}

export const favoritesService = new FavoritesService()